mypy_django_plugin/transformers/managers.py

```diff
@@ -274,6 +285,26 @@ def resolve_manager_method(ctx: AttributeContext) -> MypyType:
 
     if isinstance(ctx.type, Instance):
         return resolve_manager_method_from_instance(instance=ctx.type, method_name=method_name, ctx=ctx)
+    elif isinstance(ctx.type, UnionType):
+        if any(isinstance(instance, AnyType) for instance in ctx.type.items):
+            ctx.api.fail(
+                f'Unable to resolve return type of queryset/manager method "{method_name}" on a union involving Any',
+                ctx.context,
+            )
+            return AnyType(TypeOfAny.from_error)
+        elif any(not isinstance(instance, Instance) for instance in ctx.type.items):
+            ctx.api.fail(
+                f'Unable to resolve return type of queryset/manager method "{method_name}"',
+                ctx.context,
+            )
+            return AnyType(TypeOfAny.from_error)
+        else:
+            resolved = tuple(
+                resolve_manager_method_from_instance(instance=instance, method_name=method_name, ctx=ctx)
+                for instance in ctx.type.items
+                if isinstance(instance, Instance)
+            )
+            return get_proper_type(UnionType(resolved))
```

Comment:
Since this now only requires `Instance`(?) I think we can switch to `if all(...)` and drop the outermost `else`

Reply:
I kept the difference because when there is an Any, the error message is more precise. I don't have a strong opinion though, I can collapse that

Reply:
fixed, and I got all tests to pass. For real this time!